cli/resolver.rs

```diff
@@ -837,19 +861,50 @@ impl NpmResolver for CliGraphResolver {
     }
   }
 
-  fn resolve_npm(&self, package_req: &PackageReq) -> NpmPackageReqResolution {
+  async fn resolve_pkg_reqs(
+    &self,
+    package_reqs: &[&PackageReq],
+  ) -> NpmPackageReqsResolution {
     match &self.npm_resolver {
-      Some(npm_resolver) => match npm_resolver.as_inner() {
-        InnerCliNpmResolverRef::Managed(npm_resolver) => {
-          npm_resolver.resolve_npm_for_deno_graph(package_req)
+      Some(npm_resolver) => {
+        match npm_resolver.as_inner() {
+          InnerCliNpmResolverRef::Managed(npm_resolver) => {
+            let reqs_with_package_infos = {
+              let memory_cache = self.memory_cache.borrow();
+              package_reqs
```

Comment:
Nit, this becomes really hard to parse, consider doing `let Some(npm_resolver) = ... else {}` and returning early where possible. Not a blocker to landing though